### PR TITLE
Added '@Botkube commands list' to show all the supported kubectl cmds 

### PR DESCRIPTION
**Pull request by girishg4t**
##### ISSUE TYPE
 - Feature Pull Request

##### SUMMARY
Added new botkube command to list all the supported commands, @botkube commands list

Below are the changes
made changes in executor.go to handle newly added command  
added common method in utils.go to format the allowed commands  

Fixes https://github.com/infracloudio/botkube/issues/312

### CODE REVIEW

File: pkg/execute/executor.go [hunk 1]
```diff
@@ -146,6 +149,14 @@ const (
 	FilterDisable FiltersAction = "disable"
 )
 
+// InfoAction for options in Info commands
+type InfoAction string
```
[the remark this answers is not in the document]
Comment (girishg4t) [reply pkg/execute/executor.go:153]: corrected `InfoAction` to `infoAction`

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -477,3 +479,15 @@ func GetClusterNameFromKubectlCmd(cmd string) string {
 	}
 	return s
 }
+
+//GetFormatedCommandsList get the formated commands list
+func GetFormatedCommandsList(header string, commands map[string]bool) string {
```
[the remark this answers is not in the document]
Comment (girishg4t) [reply pkg/utils/utils.go:484]: corrected name to `GetStingInYamlFormat`, pls suggest

File: test/e2e/command/botkube.go [hunk 3]
```diff
@@ -90,6 +115,7 @@ func (c *context) testBotkubeCommand(t *testing.T) {
 
 func compareFilters(expected, actual []string) bool {
 	if len(expected) != len(actual) {
+		log.Infof("************************Expected %d Actual %d", len(expected), len(actual))
```
[the remark this answers is not in the document]
Comment (girishg4t) [reply test/e2e/command/botkube.go:118]: Removed

File: pkg/execute/executor.go [hunk 4]
```diff
@@ -146,6 +149,14 @@ const (
 	FilterDisable FiltersAction = "disable"
 )
 
+// infoAction for options in Info commands
+type infoAction string
+
+// Info command options
+const (
+	InfoList infoAction = "list"
```
Comment (PrasadG193) [inline pkg/execute/executor.go:157]: Avoid exporting variables if not required

Comment (girishg4t) [reply pkg/execute/executor.go:157]: done

File: pkg/execute/executor.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (girishg4t) [reply pkg/execute/executor.go]: The output of the command is
```
allowed verbs:
- cluster-info
- diff
- get
- api-resources
- api-versions
- describe
- explain
- logs
- top
- auth
allowed resources:
- storageclasses
- nodes
- deployments
- pods
- namespaces
- daemonsets
- statefulsets
```

File: pkg/execute/executor.go (hunk 4 above)
Comment (girishg4t) [reply pkg/execute/executor.go]: Done

**Review by PrasadG193 — APPROVED** [review]

File: pkg/utils/utils.go [hunk 5]
```diff
@@ -477,3 +479,15 @@ func GetClusterNameFromKubectlCmd(cmd string) string {
 	}
 	return s
 }
+
+//GetStingInYamlFormat get the formated commands list
+func GetStingInYamlFormat(header string, commands map[string]bool) string {
```
Comment (PrasadG193) [inline pkg/utils/utils.go:484]:
```suggestion
func GetStringInYamlFormat(header string, commands map[string]bool) string {
```

Comment (PrasadG193) [reply pkg/utils/utils.go:484]: Typo :)

**Review by PrasadG193 — APPROVED** [review]

File: pkg/utils/utils.go (hunk 5 above)
Comment (girishg4t) [reply pkg/utils/utils.go]: Corrected thanks

**Pull request merged**